modules/doc/content/getting_started/examples_and_tutorials/tutorial01_app_development/step06_input_params.md

```diff
@@ -1,6 +1,152 @@
+
+## Demonstration id=demo
+
+Recall from the [previous step](tutorial01_app_development/step05_kernel_object.md#physics) that, upon applying the [!ac](BVP), the weak form of Darcy's pressure equation is
+
+!equation id=darcy-weak
+(\nabla \psi, \dfrac{K}{\mu_{f}} \nabla p) = 0
+
+Also, recall that the property values are $K = 0.8451 \times 10^{-9} \, \textrm{m}^{2}$ and $\mu_{f} = 7.98 \times 10^{-4} \, \textrm{Pa} \cdot \textrm{s}$.
+
+### Source Code id=source-demo
+
+The `DarcyPressure` object shall be modified to allow users to define the constant properties. A required parameter, `"permeability"`, shall be used to set the value of $K$ and an optional parameter, `"viscosity"`, whose default value is the viscosity of water at $30 \degree \textrm{C}$, shall be used to set the value of $\mu_{f}$. Assuming that $K$ and $\mu_{f}$ are valid for any number in $\mathbb{R}$, the data type of these parameters should be `Real`. The only exception is that $\mu_{f}$ cannot be zero, since this would produce a `NaN` value. To enforce this constraint, the parameter object shall be created with `addRangeCheckedParam()`, which is similar to `addParam()`, but it accepts a parsed logical expression that must evaluate to true.
+
+In `DarcyPressure.C`, change the `validParams()` definition to the following:
+
+!listing tutorials/tutorial01_app_development/step06_input_params/src/kernels/DarcyPressure.C
+         link=False
+         start=InputParameters
+         end=DarcyPressure::DarcyPressure(const InputParameters & parameters)
+
+The parameters will be retrieved by the constructor method and used to set the values for the `_permeability` and `_viscosity` variables. The user-inputs shall be read with the `getParam()` method. Thus, change the constructor definition to the following:
+
+!listing tutorials/tutorial01_app_development/step06_input_params/src/kernels/DarcyPressure.C
+         link=False
+         start=DarcyPressure::DarcyPressure(const InputParameters & parameters)
+         end=}
+         include-end=True
+
+Do not modify any other parts of `DarcyPressure.C`. Now, recompile the application:
+
+```bash
+cd ~/projects/babbler
+make -j4
+```
+
+### Input File id=input-demo
+
+The properties in [darcy-weak] can now be specified in the input file. However, since the default for the `"viscosity"` parameter is the desired value of $\mu_{f} = 7.98 \times 10^{-4} \, \textrm{Pa} \cdot \textrm{s}$, it need not be set. The `[Kernels]` block in `pressure_diffusion.i` should be as follows:
+
+!listing tutorials/tutorial01_app_development/step06_input_params/problems/pressure_diffusion.i
+         block=Kernels
+         link=False
+
+Now, execute the input file:
+
+```bash
+cd ~/projects/babbler/problems
+../babbler-opt -i pressure_diffusion.i
+```
+
+!alert! tip title=Try to input a zero-valued viscosity.<!-- this will be a good demo for RunException on the next step about unit tests-->
```

Comment:
```suggestion
!alert! tip title=Try to input a zero-valued viscosity. !!this will be a good demo for RunException on the next step about unit tests
```

Reply:
In the next step, can you also add in a manual check in the constructor for the permeability parameter to be create than zero to introduced paramError/mooseError. Then create the test for it.

```
if (_permeabiliy < 0)
  paramError(...)
```

Reply:
Seems a bit redundant, no? Perhaps at some other point later in the tutorial there will be a better opportunity to demonstrate this?